packages/@aws-cdk/lambda-layer-awscli/lib/awscli-layer.ts

```diff
@@ -1,19 +1,150 @@
+/* eslint-disable no-console */
+import * as childproc from 'child_process';
+import * as fs from 'fs';
 import * as path from 'path';
 import * as lambda from '@aws-cdk/aws-lambda';
-import { FileSystem } from '@aws-cdk/core';
+import { Annotations, FileSystem } from '@aws-cdk/core';
+import { debugModeEnabled } from '@aws-cdk/core/lib/debug';
+import * as cxapi from '@aws-cdk/cx-api';
 import { Construct } from 'constructs';
 
 /**
  * An AWS Lambda layer that includes the AWS CLI.
  */
 export class AwsCliLayer extends lambda.LayerVersion {
+  /**
+   * @internal
+   */
+  public static _tryLoadPackage(targetVersion: string, logs: string[]): any {
+    let availableVersion;
+    let assetPackagePath;
+    try {
+      assetPackagePath = require.resolve(`${AwsCliLayer.assetPackageName}`);
+    } catch (err) {
+      logs.push(`require.resolve('${AwsCliLayer.assetPackageName}') failed`);
+      return;
+    }
+    availableVersion = AwsCliLayer.requireWrapper(path.join(assetPackagePath, '../../package.json'), logs).version;
+
+    if (targetVersion === availableVersion) {
+      logs.push(`${AwsCliLayer.assetPackageName} already installed with correct version: ${availableVersion}.`);
+      const result = AwsCliLayer.requireWrapper(AwsCliLayer.assetPackageName, logs);
+      if (result) {
+        logs.push(`Successfully loaded ${AwsCliLayer.assetPackageName} from pre-installed packages.`);
+        return result;
+      }
+    }
+  }
+
+  /**
+   * @internal
+   */
+  public static _downloadPackage(targetVersion: string, logs: string[]): string | undefined {
+    const cdkHomeDir = cxapi.cdkHomeDir();
+    const downloadDir = path.join(cdkHomeDir, 'npm-cache');
+    const downloadPath = path.join(downloadDir, `${AwsCliLayer.assetPackageNpmTarPrefix}${targetVersion}.tgz`);
+
+    if (fs.existsSync(downloadPath)) {
+      logs.push(`Using package archive already available at location: ${downloadPath}`);
+      return downloadPath;
+    }
+    logs.push(`Downloading package using npm pack to: ${downloadDir}`);
+    childproc.execSync(`mkdir -p ${downloadDir}; cd ${downloadDir}; npm pack ${AwsCliLayer.assetPackageName}@${targetVersion} -q`);
```

Comment:
Probably should be connected with `&&` in case the fs operations might fail. Not sure if they would, but they might.

Also this is not (necessarily) going to work on Windows, come to think of it? Better do `mkdir` in Node, and run `npm pack` with a `cwd`.

Reply:
Ah yes, we need to be platform independent.